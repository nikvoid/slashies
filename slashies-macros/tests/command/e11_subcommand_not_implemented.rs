use slashies_macros::Command;

/// An invalid command
#[derive(Command)]
#[name = "BadCommand"]
enum BadCommand {
    /// A variant that doe snot implement SubCommand
    #[name = "BadSubcommand"]
    Sub(SubCommand),
}

struct SubCommand;

fn main() {}
