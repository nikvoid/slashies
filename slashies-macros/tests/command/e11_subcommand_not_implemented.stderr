error[E0277]: the trait bound `BadCommand: ApplicationCommandInteractionHandler` is not satisfied
   --> tests/command/e11_subcommand_not_implemented.rs:4:10
    |
4   | #[derive(Command)]
    |          ^^^^^^^ the trait `ApplicationCommandInteractionHandler` is not implemented for `BadCommand`
    |
note: required by a bound in `slashies::Command`
   --> $WORKSPACE/slashies/src/lib.rs
    |
    | pub trait Command: ApplicationCommandInteractionHandler + Sized {
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `slashies::Command`
    = note: this error originates in the derive macro `Command` (in Nightly builds, run with -Z macro-backtrace for more info)
